functions/go/set-setter/set.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+var _ kio.Filter = Set{}
+
+// Set sets resource field values from an OpenAPI setter
+type Set struct {
+	// Setters holds the user provided values for all the setters
+	Setters []Setter `json:"setters,omitempty" yaml:"setters,omitempty"`
+}
+
+type Setter struct {
+	// SetterName is the name of the setter
+	SetterName string `json:"setter_name,omitempty" yaml:"setter_name,omitempty"`
+
+	// SetterValue is the input value for setter
+	SetterValue interface{} `json:"setter_value,omitempty" yaml:"setter_value,omitempty"`
+}
+
+const SetPrefix = "# kpt-set: "
+
+// Filter implements Set as a yaml.Filter
+func (s Set) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	for i := range nodes {
+		err := accept(&s, nodes[i])
+		if err != nil {
+			return nil, errors.Wrap(err)
+		}
+	}
+	return nodes, nil
+}
+
+func (s *Set) visitMapping(_ *yaml.RNode, p string) error {
```

Comment:
Are these 2 functions expected to do nothing eventually?
If we plan to add something in these 2 functions, let's add a TODO.

Reply:
Added comments.